Integrated MockAPI within the map logic

Colividiu opened the pull request:
The coordinates are now fetched and updated using http methods. The drones' movements are still faked by timed increments, however at a lower speed (1 sec), in order to prevent errors concerning too many http requests being sent

Colividiu committed:
Integrated MockAPI within the map logic

The coordinates are now fetched and updated using http methods. The drones' movements are still faked by timed increments, however at a lower speed (1 sec), in order to prevent errors concerning too many http requests being sent

tomoiagahoratiu commented on src/components/MapAP.tsx:
```diff
@@ -1,45 +1,55 @@
 import '../styles/Map.css'
 import logo from '../assets/LogoCC.png';
 import { Drone } from '../models/Drone';
+import axios from 'axios';
```
you need to add the dependency for axios

tomoiagahoratiu commented on src/components/MapAP.tsx:
```diff
@@ -1,45 +1,55 @@
+      const response = await axios.get(`${API_BASE_URL}/Drone`);
+      setDronesList(response.data);
+    } catch (error) {
+      console.error("Error fetching drones data:", error);
```
we should remove all the console logs, and maybe create a scenario for the failing call

Colividiu committed:
Added missing dependencies

Colividiu committed:
Added scenarios for possible errors when
communicating with the API

Changes were made in the logic to acommodate error handling (for ex the coords stop incrementing after 5 unsuccessful attempts of updating them using axios.put). Unnecessary console.logs were also removed here and the code is overall more stable than before (the update interval can be set to lower times without encountering errors from oversoliciting the API

Colividiu committed:
Color coded the map markers/Parked drones remain stationary

Colividiu commented:
I made the changes we discussed and did the next task

Colividiu committed:
Moved the map marker constants into the utils directory

Cleaned up the structure in the models directory a little bit, now it only includes types

tomoiagahoratiu commented:
good job 👍
